src/transformers/models/deberta/modeling_deberta.py

```diff
@@ -773,6 +775,18 @@ def _init_weights(self, module):
         if isinstance(module, nn.Linear) and module.bias is not None:
             module.bias.data.zero_()
 
+    def _pre_load_hook(self, state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs):
+        self_state = self.state_dict()
+        if (
+            ("classifier.weight" in self_state)
+            and ("classifier.weight" in state_dict)
+            and self_state["classifier.weight"].size() != state_dict["classifier.weight"].size()
+        ):
+            logger.warning("Ignore mismatched classifer head.")
```

Comment:
The warning could be made better: 
```suggestion
            logger.warning(f"The checkpoint classifier head has a shape {state_dict["classifier.weight"].size()} and this model classifer head has a shape {self_state["classifier.weight"].size()}. Ignoring the checkpoint weights. You should train your model on new data.")
```